orderbook/openapi.yml

```diff
@@ -240,6 +240,41 @@ paths:
                 $ref: "#/components/schemas/FeeInformation"
         404:
           description: Token non-existent or not connected to native token
+  /api/v1/price/{baseToken}/{quoteToken}:
+    get:
+      description: |
+        The estimated price for selling an optionally specified amount of baseToken or buying that amount of quote token. Price is denominated in quote token.
+        E.g. the price for WETH/USDC (0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2/0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48) might be around 1800 USDC.
+      parameters:
+        - name: baseToken
+          in: path
+          required: true
+          schema:
+            $ref: "#/components/schemas/Address"
+        - name: quoteToken
+          in: path
+          required: true
+          schema:
+            $ref: "#/components/schemas/Address"
+        - name: amount
+          in: query
+          schema:
+            $ref: "#/components/schemas/TokenAmount"
+            default: "0"
```

Comment:
What would this mean, to sell 0 tokens?

Reply:
It would mean to get the spot price. I specified defaults so that the endpoint without any arguments can be used to estimate the spot price. But we can make everything mandatory and expose a separate spot price route.

Reply:
For me this behavior would be ok, and we can leave it. Maybe It would be good to mention in the description if we want to keep it like this.

I have some misconception about prices, maybe it would help if we have a glossary that we agree on with the different relevant prices. Let me explain. A price can be for example "the best ask", "best buy", the "mid point of those", or it can be "amount dependant", or even it can be defined as the "last buy", or the "last sell". 
So, it's hard to talk about prices as an absolute number. 

My understanding is, all those prices will have "different names", and I'm not the best at choosing those, but I think doing so wisely is important. So my point is. should we ask for this amount to be mandatory and positive, and use different "endpoint name" for the price that has 0 as the amount, and this endpoint?

I think it would help to start giving names to these prices, and decide the ones we want to do now.

Reply:
If we force amount > 0 and not have a default order kind, this should maybe not return a price but an amount instead? This way we can do things like `/api/v1/markets/{baseToken}-{quoteToken}/sell/1000`

Reply:
that's a convenient endpoint yep, to know how much you are getting. Although, I would still add one for prices. Even if they are different ways of seeing the same info

But maybe what's best, is to check how other APIs return the price. And choose one we think is good :)

Reply:
Also, other alternative is:

`/api/v1/markets/{baseToken}-{quoteToken}/sell-price?amount=1000`

returns:
```js
{
   price: '125.2345678',
   sellAmount: '1000',
   buyAmount: '125234.5678'
}
```
I think this would be quite easy to read

Reply:
The thing I don't like about the price + amount together endpoint is that it would require two units. Sell/buy token and price currency. Turning a buy and sell amount into a price is something every client should be able to do locally, and we can expose best bid/best ask/price endpoints later on.